proof.go

```diff
@@ -528,6 +528,10 @@ func (proof Proof) VerifySubtreeRootInclusion(nth *NmtHasher, subtreeRoots [][]b
 			return popIfNonEmpty(&subtreeRoots), nil
 		}
 
+		if end-start == 1 {
```

Comment:
[question] what is the intuition around end-start == 1? If end is non-inclusive and start is inclusive and I want to refer to a single node (i.e. node at index 3) then I assume start=3, end=4. In this case end-start does = 1.

Reply:
I guess I need to add a comment there to explain it more. 
At that level, we reached a leaf, and we looked in the subtree roots whether we have it, and also looked in the side nodes if we have it, but we can't find any node. This means that since the begining, the provided ranges do not reference valid inner nodes

Reply:
ok thanks!

Reply:
It would be nice if a comment was added here to explain the logic

Reply:
https://github.com/celestiaorg/nmt/pull/272